### ISSUE TITLE
Propose to remove bold from shadow titles

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#1205]
![unjustified bold](https://cloud.githubusercontent.com/assets/4534056/5331239/8fef375a-7e21-11e4-9ef3-54c7b3d49122.png)

As far as I can tell, the bold is not justified and not pretty. If I understand it right it's to show they're shadows, but is this important to know in the sidebar - or, rather, in tab _Open_ where it's one of the few places shadow titles appear... other than in tab _Shadow_... where actually the bold also adds clutter. The $ indicates this is a shadown anyway and if you try to edit it you get a clear warning.

...

**ADDENDUM, years later**, after discussions that have futher clarified the problem:

The different styling is not information **about the link** but rather information about the *tiddler*. The current styling at most pre-informs you that it is a shadow (that is not overwritten) but I fail to see why this is so important to know at this stage, i.e when merely glancing the title - to *do* something with the tiddler you'll need to *click* the link. The status of "shadow or not" is really about the *tiddler*, not about the *link*.

A fair exception for link styling is *missing* tiddlers, so you are informed that you perhaps shouldn't click the link at all. That **is** information about the link, i.e what to *do* with the link! 

**The multiple native link styles bring clutter** when the supposed information given from the styling is not needed or requested. (But I wouldn't mind some kind of tooltip or other *opt-in* to get more tiddler info.)

However, a perhaps separate issue, is that the shadow **tiddlers** themselves ought to clearly inform what they are, as is done e.g when clicking to edit them or when they're being edited.

**Comment by Leilei332** [comment]
Related issue: #8711 

Using another color for links to shadow tiddlers is still unusable for people who have difficulty identifying colors. IMO it may be better to use borders to distinguish shadows and overriden shadows (like dashed border for plain shadow and solid border for overriden shadow).

<img width="1396" height="640" alt="Image" src="https://github.com/user-attachments/assets/82a12f55-90a0-47e4-84ce-70878ab42fc8" />

**Comment by pmario** [comment]
I am not happy with a border all around the link, **but** on one side eg: left it would be OK for me

**Comment by yaisog** [comment]
Please not on the left side.
In the last 20 years or so we were conditioned to recognize links by an underline (not a bottom border). Don't break with that, please. Can't we have dotted underline maybe?
What happened to the agreement that bold should be used for overwritten shadows, the result of the lengthy discussion above?

**Comment by Leilei332** [comment]
> Please not on the left side. In the last 20 years or so we were conditioned to recognize links by an underline (not a bottom border). Don't break with that, please. Can't we have dotted underline maybe?

IMO `border-bottom` can be a candidate, since it looks similar to an underline. The problem of `text-decoration` is that its line is too thin to see, and `text-decoration-thickness` requires 2021 baseline.

<img width="1396" height="640" alt="Image" src="https://github.com/user-attachments/assets/f2120371-c4b8-456a-95ca-ed2f6b613bfd" />

> What happened to the agreement that bold should be used for overwritten shadows, the result of the lengthy discussion above?

IMO we should emphasize overridden shadow tiddlers, since the number of overridden tiddlers is less than ordinary shadow tiddlers.

**Comment by Leilei332** [comment]
Another possible candidate: display a unicode symbol on the left.

<img width="1396" height="706" alt="Image" src="https://github.com/user-attachments/assets/6538afec-f188-46d0-b4b1-6ccc6a8e2cc5" />

**Comment by twMat** [comment]
@pmario @yaisog @Leilei332 @Jermolene 

We're missing out on a most important point that @Jermolene makes (at bottom of [this](https://github.com/TiddlyWiki/TiddlyWiki5/issues/1205#issuecomment-66641227) post):

> I agree that making shadow links visually distinctive is irrelevant for many users. I wouldn't object to removing the special styling (but perhaps we should make shadow tiddlers themselves visually distinctive, as we do with system and missing tiddlers).

So, instead of all these (still cluttery!) proposals, let's just skip the special styling and make the shadow tiddlers themselves visually distinctive. Here are some ideas:

<img width="608" height="197" alt="Image" src="https://github.com/user-attachments/assets/fb01af08-dc53-49f9-be46-33150a5060ff" />

<br> This is a shadowbox inset on the tiddler frame:
<img width="608" height="82" alt="Image" src="https://github.com/user-attachments/assets/48d6ba66-da24-4046-9224-329bbc59f60b" />

<br>But, frankly, **no** styling makes it obvious that it means the tiddler is a shadow! I think the following approach would be more useful, i.e to indicate that the tiddler actually "belongs" to another tiddler (i.e the actual *meaning* of "shadow"):
<img width="608" height="97" alt="Image" src="https://github.com/user-attachments/assets/704f29b3-350d-420e-8036-75815eaeb7a2" />

<br>...maybe this is even an opportunity to visually simplify complex titles and foster use of name spaces? - 
<img width="604" height="115" alt="Image" src="https://github.com/user-attachments/assets/9b14ebcd-e070-4c73-999f-22d3fd7176a1" />

<br>...or...
<img width="608" height="116" alt="Image" src="https://github.com/user-attachments/assets/326282b9-b0ef-4604-bbc6-e70d696923ac" />

**Comment by pmario** [comment]
>So, instead of all these (still cluttery!) proposals, let's just skip the special styling and make the shadow tiddlers themselves visually distinctive. 

That's right. But why do I need to open the tiddler, if I can see it at the link-level. 

Most of the time I do not care, if a tiddler is a shadow tiddler. Except when I do care. -- Sometimes.

**Comment by twMat** [comment]
> Most of the time I do not care, if a tiddler is a shadow tiddler. 

Exactly, so it should not clutter the lists. That's the main point with the OP. 

Just like for the rest of the tiddlers meta-info (and, for that matter, its actual info) you *simply* open the tiddler if you want to know such stuff.

(...or how about we style all kinds of meta-information into the link!? All tiddler *types* could have individual link styles, or whether a tiddler is also a tag or not, or whether it is tagged `$:/tags/Global`/`Macro`, maybe if it has more than 42 characters in it or not, ...  it will be Christmas trees in the sidebar all year around ;-)

**Comment by pmario** [comment]
For exactly that reason our links do have different class values assigned. So users are free to style their links in any way they want. 

Currently we do have an indicator for overwritten shadow tiddlers. IMO that **is** an information, that should stand out. 

Currently a shadow, that is also a tiddler uses this CSS:

```
a.tc-tiddlylink-shadow.tc-tiddlylink-resolves {
  font-weight: normal;
}
```

Standard shadow tiddlers are bold:

```
a.tc-tiddlylink-shadow {
  font-weight: bold;
}
```

We could turn the default behaviour around, so .tc-tiddlylink-resolves is **bold**

**Comment by Leilei332** [comment]
> (...or how about we style all kinds of meta-information into the link!? All tiddler *types* could have individual link styles, or whether a tiddler is also a tag or not, or whether it is tagged `$:/tags/Global`/`Macro`, maybe if it has more than 42 characters in it or not, ...  it will be Christmas trees in the sidebar all year around ;-)

I once did think it is very useful for link widget to intergrate cascades, but it may be a burden for parsers, which may impact performance.

**Comment by twMat** [comment]
Note: I've updated the OP with some clarifications.

...

> I once did think it is very useful for link widget to intergrate cascades, but it may be a burden for parsers, which may impact performance.

This is of course another matter than the OP, i.e for another thread. But, yes, I can imagine many useful things that I think would require cascades there.

**Comment by Leilei332** [comment]
I'm now thinking of a possible compromise solution: Disable link bold weight by default, a `tc-link-weight` class is required to be added to the parent element to restore the previous behaviour. Perhaps we can only enable link bold weight by default in advancedsearch and some locations.

**Comment by pmario** [comment]
`tc-link-weight` ... weight is too specific. What if later we or users want to change the color of shadow links. IMO The class definition names are fine. It's the styles, we may want to change.
